app/src/main/java/com/codingblocks/cbonlineapp/dashboard/DashboardViewModel.kt

```diff
@@ -325,4 +327,71 @@ class DashboardViewModel(
     }
 
     fun getPerformance(attemptId: String) = homeRepo.getRunStats(attemptId)
+
+    var wishlist = MutableLiveData<List<Wishlist>>()
+    var toastMutable = MutableLiveData<String>()
+    fun fetchWishList(){
+        runIO {
+            when (val response = homeRepo.fetchWishlist()) {
+                is ResultWrapper.GenericError -> setError(response.error)
+                is ResultWrapper.Success -> {
+                    if (response.value.isSuccessful) {
+                        wishlist.postValue(response.value.body()?.get())
+                    } else {
+                        setError(fetchError(response.value.code()))
+                    }
+                }
+            }
+        }
+    }
+
+    fun changeWishlistStatus(id: String){
+        runIO {
+            when (val response = homeRepo.checkIfWishlisted(id)) {
+                is ResultWrapper.GenericError -> setError(response.error)
+                is ResultWrapper.Success -> with(response.value) {
+                    if (isSuccessful) {
+                        if (response.value.body()?.id!=null){
+                            removeFromWishlist(response.value.body()?.id)
+                        }else{
+                            addToWishlist(id)
+                        }
+                    } else {
+                        setError(fetchError(code()))
+                    }
+                }
+            }
+        }
+    }
+
+    fun addToWishlist(id: String){
+        val wishlist = Wishlist(Course(id), User(prefs.SP_USER_ID))
```

Comment:
do we really need send user id it should take automatically from jwt token if I'm not wrong